internal/datastore/spanner/watch.go

```diff
@@ -164,3 +181,175 @@ func contextualizedCaveatFromValues(values map[string]any) (*core.Contextualized
 	}
 	return nil, nil
 }
+
+const maxSchemaWatchRetryCount = 15
+
+func (sd spannerDatastore) WatchSchema(ctx context.Context, afterRevision datastore.Revision) (<-chan *datastore.SchemaState, <-chan error) {
```

Comment:
can we add mocked tests at least if the emulator does not properly do changestreams?

Reply:
As discussed, will wait for the emulator's support to catch up